### DISCUSSION TITLE
GraphQL - Fetch M2A Relationship Item Values

### DISCUSSION THREAD

**Discussion by ddcollins** [directus/directus#3248] [Archive]
I have 3 models:
1. qualification
    - title
    - steps (M2A of which can be a "course" or "video") 
    - steps is defined by `qualification_steps` table
        - id
        - collection ("course" or "video)
        - item (id of course or video)
        - qualification_id
2. video
    - title
3. course
    - title

I am trying to write a GraphQL query to pull the qualifications, the steps, and the title of each step. However, the `title` field for each potential step is coming back as `null`. Is this unsupported without defined interfaces in the schema, or might I just be missing something?

Also notice the `__typename` of the video step (which is collection of "video") is actually "course"...

This is my GraphQL query:
```
{
    items {
        qualification(limit:10) {
            title
            steps {
                id
                collection
                item {
                    __typename
                    ... on course {
                        id
                        title
                    }
                    ... on video {
                        id
                        title
                    }
                }
            }
        }
    }
}
```

This is the result of that query:
```
{
    "data": {
        "items": {
            "qualification": [
                {
                    "title": "Level 01",
                    "steps": [
                        {
                            "id": "1",
                            "collection": "course",
                            "item": {
                                "__typename": "course",
                                "id": null,
                                "title": null
                            }
                        },
                        {
                            "id": "2",
                            "collection": "video",
                            "item": {
                                "__typename": "course",
                                "id": null,
                                "title": null
                            }
                        }
                    ]
                }
            ]
        }
    }
}
```

**Comment by evgheni7** [comment]
Same for me, i have 4 subitems with different types, but in result all of them in same type and empty values:

QUERY
`query {
  items {
    page {
      name
      layout {
        name
        __typename
        components {
          item { 
            __typename
            ... on component_text {
              text
            }
            ... on component_image {
              images {
                id
              }
            }
          }
        }
      }
    }
  }
}`

RESPONSE:
`{
  "data": {
    "items": {
      "page": [
        {
          "name": "Home Page",
          "layout": {
            "name": "Default",
            "__typename": "layout",
            "components": [
              {
                "item": {
                  "__typename": "component_text",
                  "text": null
                }
              },
              {
                "item": {
                  "__typename": "component_text",
                  "text": null
                }
              },
              {
                "item": {
                  "__typename": "component_text",
                  "text": null
                }
              },
              {
                "item": {
                  "__typename": "component_text",
                  "text": null
                }
              }
            ]
          }
        }
      ]
    }
  }
}`

Looks like bug.

**Reply by ddcollins** [reply]
It's actually due to the M2A relation type not being supported in GraphQL yet.

See here: https://github.com/directus/directus/blob/main/api/src/services/graphql.ts#L166

I spoke to @rijkvanzanten, who wrote this, and he said it's high on his priority list (but not currently being worked on), and the quickest way to get it resolved would be to contribute.

**Reply by evgheni7** [reply]
I hope is not big issue, if result returned, but with wrong data, will try to do fix.

**Reply by ddcollins** [reply]
@rijkvanzanten Has this moved further up your TODO list by chance?

**Reply by benhaynes** [reply]
[**Unfortunately, we can not provide estimates on features or fixes**](https://docs.directus.io/getting-started/support.html#can-you-give-an-eta-for-this-feature-fix). As of now, the only way to get an accurate timeline or release date is to [sponsor](https://docs.directus.io/getting-started/support.html#sponsored-work) this task. We recognize that this may be an important feature/fix, but we are a small open-source organization with a lot to triage and complete. If you can't sponsor this ticket, then the next best thing is to increase its priority by giving it a 👍 . Or contribute a pull-request... after all, this is OSS!

**Reply by evgheni7** [reply]
@benhaynes I understand this is open source project, but this is core feature you mentioned in release, how it usable after users read about it in feature list? Or remove M2A feature from features list or let's try to fix it before stable version released.

**Reply by benhaynes** [reply]
We have M2A (REST) and have GraphQL — the website is correct. We're working on feature parity between these two APIs before release of 9.0.0. I understand that some missing things are frustrating, but it's the reality of a small team trying to work as efficiently as possible.

**Reply by evgheni7** [reply]
@benhaynes I really appreciate your work, but you mention nothing about just REST implemented feature, in docs there:
https://docs.directus.io/concepts/relationships.html#many-to-any-m2a
What makes users confusing in time when you have already RC.

**Reply by benhaynes** [reply]
I get it, and apologize for the misinformation. As you can see, the Docs are still very incomplete for v9. We'll be spending time on the docs once we have our triage list of bugs and 9.0.0 features done. Bear with us while things are in flux over the next few weeks, and feel free to submit some PRs to resolve things.

**Comment by suyashnigam50** [comment]


**Reply by br41nslug** [reply]
This thread has been inactive for over 2 years. If you're running into a similar bug please do create an issue for it 😄
